bin/packages/build.js

```diff
@@ -29,41 +30,77 @@ function getPackageName( file ) {
 }
 
 /**
- * Returns a stream transform which maps an individual stylesheet to its
- * package entrypoint. Unlike JavaScript which uses an external bundler to
- * efficiently manage rebuilds by entrypoints, stylesheets are rebuilt fresh
- * in their entirety from the build script.
+ * Helper higher order function for creating stream transforms that map from
+ * one input file to none, one, or multiple output files.
+ *
+ * @param {Function} mappingFunc An async mapping function that receives a file
+ * 								 path and returns either a falsy to indicate
+ * 								 the file should not be built, a string file
+ * 								 path to indicate that this single file should
+ * 								 be built, or an array of strings to indicate
+ * 								 multiple files should be built.
  *
  * @return {Transform} Stream transform instance.
  */
-function createStyleEntryTransform() {
-	const packages = new Set;
+function createEntryTransform( mappingFunc ) {
+	// Track any already built files to avoid duplication.
+	const fileSet = new Set;
```

Comment:
The second commit in this branch introduces this generic way to avoid duplicate built files, but I'm not sure it's the right solution. 

It doesn't avoid repeating expensive operations in the `mappingFunc`, which I imagine was the original intention for tracking changed packages. (cc @aduth)

Reply:
Hm, yeah, I see what you mean. Maybe we can pass `fileSet` as the second argument in the mapping function to allow the callback to determine whether to defer to a cached value?

Or if we really want to follow `Array#map` signature, as the third argument, and likely as an array and not a set ([documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map#Syntax)).